app/ordering_components.py

```diff
@@ -42,15 +43,18 @@ def get_clinical_evidence(result, message, logger, db_conn):
     return compute_clinical_evidence(result, message, logger, db_conn)
 
 
-def get_novelty(result, message, logger):
-    # TODO get novelty from novelty package
-    return 0
+def get_novelty(message, logger):
+    return compute_novelty(message, logger)
 
 
 def get_ordering_components(message, logger):
     logger.debug(f"Computing scores for {len(message['results'])} results")
     db_conn = redis.Redis(connection_pool=redis_pool)
-    for result_index, result in enumerate(tqdm(message.get("results") or [])):
+    novelty_scores_dict = get_novelty(message, logger).to_dict(orient="index")
+    novelty_scores = {
```

Comment:
This doesn't appear used until 74. Should we move this inside of the if at 69 to reduce computation when clinical_evidence_score is non-zero?

Reply:
Moving it into the `results` for loop isn't good because this gets novelty for the entire message and is by far the slowest part of this code. I could move it to below the for loop and only calculate if there are any clinical_evidence_scores == 0, but the chances of that are so small that I don't think it matters, and then I would need to loop over all results again.